torch/__init__.py

```diff
@@ -917,6 +917,59 @@ def is_warn_always_enabled():
     """
     return _C._get_warnAlways()
 
+
+def _check_with(error_type, cond, message):
+    if not isinstance(cond, builtins.bool):
+        raise TypeError(f'cond must be a bool, but got {type(cond)}')
+
+    if cond:
+        return
+
+    # error_type must be a subclass of Exception and not subclass of Warning
+    assert issubclass(error_type, Exception) and not issubclass(error_type, Warning)
+
+    if message is None:
+        message_evaluated = (
+            'Expected cond to be True, but got False. (Could this error '
+            'message be improved? If so, please report an enhancement request '
+            'to PyTorch.)')
```

Comment:
BTW, this is not as important in Python land, as users get stack traces by default. But it's fine probably.